### DISCUSSION TITLE
Split the JavaScript course into two courses

### DISCUSSION THREAD

**Discussion by nik-rev** [TheOdinProject/curriculum#27482] [Ideas]
#### Why?

To understand why I think this would be a good idea, first let's talk about the intermediate and advanced html/css courses.

Both of these courses could very well be merged into a single course, however it is better that they're not as it gives a sort of a breather for the learner to not have to do all the html/css in a single course. They do the first half of the content, and while they're in the JavaScript section they focus on the JavaScript while (mostly) not focusing nearly as much on the HTML/CSS. 

The JavaScript course is huge. It introduces many topics, so it may be better to split it up to make it easier to digest. This way, when we've completed javascript in foundations, we do some html/css. Then, some javascript, some html/css and finally finishing off with more javascript.

For example, by the time the learner finishes off the weather app project they may be slightly exhausted from doing so much javascript, so it could be beneficial for them to take a moment to focus on html/css before diving into more javascript, then when they come back, its like a new thing for them all over again, so they are full of energy to complete the rest of the JavaScript content.

Please let me know what you think about this! Just throwing an idea out there.

##### Intermediate JavaScript
- All the content before the Weather App Project
- Comes after Intermediate HTML and CSS

##### Advanced JavaScript
- All the content after the Weather App Project
- Comes after Advanced HTML and css

##### Ruby

Maybe the Ruby course could also be split this way, first half comes before intermediate html/css and the second half comes after. However, I don't know anything about how the Ruby course is structured to have a proper opinion on this.

**Comment by mao-sz** [comment]
One thing to consider is that splits aren't necessarily entirely (if even at all) due to "providing a breather" through a different section. It can also be because certain concepts make more sense to be taught after a certain point than an earlier one e.g. after naturally getting more practice with many of the previously taught HTMl/CSS tools throughout the JS section (since even if you don't spend a lot of time styling, you will inevitably still get a fair amount of practice with many of those tools).

E.g. is it pedagogically wise to start talking about accessibility and fully responsive design *immediately* after intermediate HTML/CSS? If one was to consider it more sensible to separate because there are more useful concepts to be taught first, can that same pedagogy be applied to the JS section? As in not just splitting for a breather but does it make sense from a teaching perspective that certain parts of the JS course *should/need to* come after, say, the advanced HTML/CSS section?

**Comment by KevinMulhern** [comment]
Converted to a discussion - this a too big for us to address in an issue. 

@MaoShizhong is spot on, we put the Intermediate HTML/CSS course where its at because its needed to style the JS projects that come after.  

But, there has been some discussion within the team about reorganising the JS path. Smaller more focused courses is likely what we'll end up having eventually.
